src/pretix/control/forms/item.py

```diff
@@ -231,12 +237,26 @@ def __init__(self, **kwargs):
             else:
                 choices.append(('{}'.format(item.pk), str(item) if item.active else mark_safe(f'<strike class="text-muted">{escape(item)}</strike>')))
 
-        self.fields['itemvars'] = forms.MultipleChoiceField(
-            label=_('Products'),
-            required=True,
-            choices=choices,
-            widget=forms.CheckboxSelectMultiple
-        )
+        if multiselect:
+            self.fields['itemvars'].choices = choices
+            self.fields['itemvars'].widget = Select2ItemVarQuotaMulti(
+                attrs={
+                    'data-model-select2': 'generic',
+                    'data-select2-url': reverse('control:event.items.itemvars.select2', kwargs={
+                        'event': self.event.slug,
+                        'organizer': self.event.organizer.slug,
+                    }),
+                    'data-placeholder': _('All products')
+                },
+                choices=choices,
+            )
+        else:
+            self.fields['itemvars'] = forms.MultipleChoiceField(
+                label=_('Products'),
+                required=True,
+                choices=choices,
+                widget=forms.CheckboxSelectMultiple
+            )
```

Comment:
This feels a little redundant now. Would

```suggestion
            self.fields['itemvars'].choices = choices
            self.fields['itemvars'].widget = forms.CheckboxSelectMultiple
```
Work the same? The `choices` part could even move out of the `if/else`.